libmscore/barline.cpp

```diff
@@ -1813,4 +1815,18 @@ QString BarLine::accessibleExtraInfo() const
     }
     return rez;
 }
+
+//---------------------------------------------------------
+//   scanElements
+//---------------------------------------------------------
+
+void BarLine::scanElements(void* data, void (* func)(void*, Element*), bool all)
+{
+    if (all || visible() || score()->showInvisible()) {
+        func(data, this);
+    }
+    if (treeChildCount() > 0) {
+        Element::scanElements(data, func, all);
```

Comment:
It's actually necessary here I guess. Here's the problem: Note may or may not be a leaf node (it may have a NoteDot for example). But we always want to call `func` on it exactly once. If I remove this `if` condition from here then it may get scanned twice sometimes. (Scanning any element twice causes it to appear slightly thicker in the score view)